src/com/google/enterprise/adaptor/filenet/DocumentTraverser.java

```diff
@@ -348,25 +368,23 @@ private void processInheritedPermissions(DocId docId,
 
   private Acl createAcl(DocId docId, Permissions.Acl permissions,
       PermissionSource permSrc, String parentFragment) {
-    Set<String> allowUsers = permissions.getAllowUsers(permSrc);
-    Set<String> denyUsers = permissions.getDenyUsers(permSrc);
-    Set<String> allowGroups = permissions.getAllowGroups(permSrc);
-    Set<String> denyGroups = permissions.getDenyGroups(permSrc);
-    if (allowUsers.isEmpty() && denyUsers.isEmpty()
-        && allowGroups.isEmpty() && denyGroups.isEmpty()) {
-      return null;
-    } else {
-      return createAcl(docId, parentFragment,
-          Acl.InheritanceType.CHILD_OVERRIDES,
-          options.getGlobalNamespace(), allowUsers, denyUsers,
-          allowGroups, denyGroups);
-    }
+    return createAcl(docId, parentFragment,
+        Acl.InheritanceType.CHILD_OVERRIDES,
+        options.getGlobalNamespace(),
+        permissions.getAllowUsers(permSrc),
+        permissions.getDenyUsers(permSrc),
+        permissions.getAllowGroups(permSrc),
+        permissions.getDenyGroups(permSrc));
   }
 
   private Acl createAcl(DocId docId, String parentFragment,
       Acl.InheritanceType inheritanceType, String namespace,
       Set<String> allowUsers, Set<String> denyUsers,
       Set<String> allowGroups, Set<String> denyGroups) {
+    if (allowUsers.isEmpty() && denyUsers.isEmpty()
+        && allowGroups.isEmpty() && denyGroups.isEmpty()) {
+      return null;
```

Comment:
I don't think this is right for marking sets or documents. For marking sets, no principals in the permissions means no access, not a null ACL, and for documents, we still need to set the inherit-from.